extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/CharChunkInputStreamGenerator.java

```diff
@@ -173,6 +191,38 @@ static WritableChunk<Values> extractChunkFromInputStream(
                 totalRows);
     }
 
+    static <T> WritableObjectChunk<T, Values> extractChunkFromInputStreamWithTransform(
+            final int elementSize,
+            final StreamReaderOptions options,
+            final Function<Character, T> transform,
+            final Iterator<FieldNodeInfo> fieldNodeIter,
+            final PrimitiveIterator.OfLong bufferInfoIter,
+            final DataInput is,
+            final WritableChunk<Values> outChunk,
+            final int outOffset,
+            final int totalRows) throws IOException {
+
+        try (final WritableCharChunk<Values> inner = extractChunkFromInputStream(
+                elementSize, options, fieldNodeIter, bufferInfoIter, is, null, 0, 0).asWritableCharChunk()) {
+
+            final WritableObjectChunk<T, Values> chunk;
+            if (outChunk != null) {
+                chunk = outChunk.asWritableObjectChunk();
+            } else {
+                final int numRows = Math.max(totalRows, inner.size());
+                chunk = WritableObjectChunk.makeWritableChunk(numRows);
+                chunk.setSize(numRows);
+            }
+
+            for (int ii = 0; ii < inner.size(); ++ii) {
```

Comment:
Should we assert that `outOffset == 0` if `outChunk == null`? Or make the allocating version delegate to the other? May be impossible due to the iteration design.